cucumber-java/src/codegen/java/CodeGenerationJava.java

```diff
@@ -0,0 +1,118 @@
+                            .filter(it -> !it.contains(String.valueOf('*')))
+                            .filter(it -> !it.matches("^\\d.*")).distinct()
+                            .forEach(keyword -> {
+                                String normalizedKeyword = normalize(keyword.replaceAll("[\\s',!\u00AD]", ""));
+                                LinkedHashMap<String, String> binding = new LinkedHashMap<>();
+                                binding.put("lang", normalizedLanguage);
+                                binding.put("kw", normalizedKeyword);
+                                try {
+                                    createInterfaces(templateSource, normalizedLanguage, normalizedKeyword, binding);
+                                } catch (IOException | TemplateException e) {
+                                    throw new RuntimeException(e);
+                                }
+                            });
+
+                    // Add package-info.java
+                    String name = dialect.getName() + ((dialect.getName().equals(dialect.getNativeName())) ? ""
+                            : " - " + dialect.getNativeName());
+                    LinkedHashMap<String, String> binding = new LinkedHashMap<>();
+                    binding.put("normalized_language", normalizedLanguage);
+                    binding.put("language_name", name);
+                    try {
+                        createPackageInfo(packageInfoSource, normalizedLanguage, binding);
+                    } catch (IOException | TemplateException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            });
+    }
+
+    private static void createInterfaces(
+            Template templateSource, String normalizedLanguage, String normalizedKeyword,
+            LinkedHashMap<String, String> binding
+    ) throws IOException, TemplateException {
+        Path path = Paths.get(baseDirectory, packagePath, normalizedLanguage, normalizedKeyword + ".java");
+        if (!Files.exists(path)) {
+            // Haitian has two translations that only differ by case - Sipozeke
+            // and SipozeKe
+            // Some file systems are unable to distinguish between them and
+            // overwrite the other one :-(
+            Files.createDirectories(path.getParent());
+            BufferedWriter writer = Files.newBufferedWriter(path, StandardOpenOption.CREATE);
+            templateSource.process(binding, writer);
+        }
+    }
+
+    private static void createPackageInfo(
+            Template packageInfoSource, String normalizedLanguage, LinkedHashMap<String, String> binding
+    ) throws IOException, TemplateException {
+        Path path = Paths.get(baseDirectory, packagePath, normalizedLanguage, "package-info.java");
+        Files.createDirectories(path.getParent());
+        BufferedWriter writer = Files.newBufferedWriter(path, StandardOpenOption.CREATE);
```

Comment:
These lines are all java `nio`. It's a much nicer interface to use then `new File()`. Might be good to learn a bit about.